openedx/openedx-platform
# feat: support unit preview in learning MFE

## Description

This PR re-implements #35687 and fixes the bug where units were erroring when the user. Additionally, this PR add support to fetch the draft branch for the sequence metadata API.

## Supporting information

See PR #35687 for more information

## Testing instructions

1. Open a course in Studio
2. Publish an existing unit
3. Make a change to the unit so that it is showing a draft version in Studio
4. Click "View live"
5. Confirm that the published version of unit is visible
6. Confirm that preview is not in the URL
7. Go back to the Studio page
8. Click "Preview"
9. Confirm that the draft version of the unit is visible
10. Confirm that "preview" is at the beginning of the URL

Repeat steps above when masquerading as a verified learner and audit learner. Confirm that the preview is displayed as expected for each track.

## Deadline

None
